File: test/fixtures/packaged/reader.js
// looks-clean — material for the packaged layer, JavaScript half.
//
// Four reads of one family so a population and a convention both exist, and
// one of them breaking the convention so the run is not empty. An empty run
// would prove the package installs, and nothing about whether it works.
import { readFile } from 'node:fs/promises';

export async function loadConfig(path) {
  try {
    return await readFile(path, 'utf8');
  } catch (e) {
    console.error('config unreadable', e);
    return null;
  }
}

export async function loadTheme(path) {
  try {
    return await readFile(path, 'utf8');
  } catch (e) {
    console.error('theme unreadable', e);
    return null;
  }
}

export async function loadLocale(path) {
  try {
    return await readFile(path, 'utf8');
  } catch (e) {
    console.error('locale unreadable', e);
    return null;
  }
}

// The odd one out: the same read, answering the same thing, with nothing left
// behind to say a failure happened.
export async function loadKeymap(path) {
  try {
    return await readFile(path, 'utf8');
  } catch {
    return null;
  }
}

File: test/fixtures/packaged/reader.dart
// looks-clean — material for the packaged layer, Dart half.
//
// THIS FILE IS THE POINT OF THE LAYER. It can only be read if the vendored
// grammar travelled inside the package, so a `files` list that forgets
// vendor/ turns this half into a parse that never happens — and the tool
// would report a clean tree rather than an error.
import 'dart:io';

class Reader {
  Future<String?> loadConfig(String path) async {
    try {
      return await File(path).readAsString();
    } catch (e) {
      print('config unreadable: $e');
      return null;
    }
  }

  Future<String?> loadTheme(String path) async {
    try {
      return await File(path).readAsString();
    } catch (e) {
      print('theme unreadable: $e');
      return null;
    }
  }

  Future<String?> loadLocale(String path) async {
    try {
      return await File(path).readAsString();
    } catch (e) {
      print('locale unreadable: $e');
      return null;
    }
  }

  // The odd one out, as above.
  Future<String?> loadKeymap(String path) async {
    try {
      return await File(path).readAsString();
    } catch (_) {
      return null;
    }
  }
}
